### PR TITLE
fix: passive support check cause memory leak

### LINKED ISSUE #20903: rc-drawer makes memory leak
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/shy-bash-vvw95)

### Steps to reproduce
Click the button and rerender the Drawer component.

### What is expected?
No detached html canvas dom

### What is actually happening?
The detached canvas dom make memory leak

| Environment | Info |
|---|---|
| antd | 3.23.6 |
| React | 16.7.0 |
| System | Window10 17763.864 |
| Browser | Chrome  78.0.3904.108 |

image

[rc-drawer relative bug](https://github.com/react-component/drawer/issues/125)

### PR DESCRIPTION
**Pull request by shaodahong**
Fixes https://github.com/ant-design/ant-design/issues/20903

### CODE REVIEW

File: src/DrawerChild.tsx [hunk 1]
```diff
@@ -81,20 +83,22 @@ class DrawerChild extends React.Component<IDrawerChildProps, IState> {
...
         Object.defineProperty({}, 'passive', {
           get: () => {
+            console.log(1123)
             passiveSupported = true;
             return null;
           },
         }),
       );
-      this.passive = passiveSupported ? { passive: false } : false;
+      window.removeEventListener("testPassive", null);
```
Comment (afc163) [inline src/DrawerChild.tsx]:
```suggestion
      window.removeEventListener('testPassive', null);
```

eslint 没生效？

File: src/DrawerChild.tsx [hunk 2]
```diff
@@ -81,20 +83,22 @@ class DrawerChild extends React.Component<IDrawerChildProps, IState> {
...
+    if (!windowIsUndefined && passiveSupported === undefined) {
+      passiveSupported = false;
       window.addEventListener(
-        'test',
-        () => { },
+        'testPassive',
+        null,
         Object.defineProperty({}, 'passive', {
           get: () => {
+            console.log(1123)
```
Comment (afc163) [inline src/DrawerChild.tsx]: 不应该有。

File: src/DrawerChild.tsx [hunk 3]
```diff
@@ -27,6 +27,8 @@ interface IState {
   prevProps?: IDrawerChildProps;
 }
 
+let passiveSupported
```
Comment (afc163) [inline src/DrawerChild.tsx]: 少 ;

File: src/DrawerChild.tsx (hunk 1 above)
Comment (shaodahong) [reply src/DrawerChild.tsx]: image

钩子冲突了

Comment (afc163) [reply src/DrawerChild.tsx]: hooks 也没生效，ci 没生效。

**Comment by yoyo837** [comment]
是不是已经通过 https://github.com/react-component/drawer/commit/e52ef83766221f94d48216c153dcd9184383b782 修复了？

**Pull request closed without merging**